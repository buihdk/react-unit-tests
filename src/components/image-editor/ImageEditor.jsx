import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import I18n from 'i18n-js';
import Cropper from 'react-cropper';
import { Modal, Button } from 'antd';

import { useIsMounted, useStateAfterMounted } from 'src/hooks';
import { noop } from 'src/utils';
import {
  previewImageProps,
  DEFAULT_FOOD_BANNER_IMAGE_RATIO,
} from 'src/config/constants';

import {
  useLoadImage,
  useHandleCancel,
  useHandleConfirm,
} from './ImageEditor.hooks';

const styles = {};

const minCropSize = 40;

const ImageEditor = ({ src, onCropImage, aspectRatio, creativeType }) => {
  const cropperRef = useRef();

  const isMounted = useIsMounted();
  const [loading, setLoading] = useStateAfterMounted(false, isMounted);
  const [cropperStyle, setCropperStyle] = useStateAfterMounted({}, isMounted);
  const [visible, setVisible] = useStateAfterMounted(false, isMounted);
  const [imageDimensions, setImageDimensions] = useStateAfterMounted(
    { width: 0, height: 0 },
    isMounted,
  );
  const [
    foodBannerAspectRatio,
    setFoodBannerAspectRatio,
  ] = useStateAfterMounted(DEFAULT_FOOD_BANNER_IMAGE_RATIO, isMounted); // 700x280

  const handleCancel = useHandleCancel({ setVisible });
  const handleConfirm = useHandleConfirm({
    setVisible,
    cropperRef,
    setLoading,
    onCropImage,
    imageDimensions,
  });

  useLoadImage({
    src,
    setVisible,
    setCropperStyle,
    setImageDimensions,
    setFoodBannerAspectRatio,
    creativeType,
  });

  return (
    <Modal
      {...previewImageProps}
      wrapClassName={styles.cropWrapper}
      centered
      width="100%"
      footer={null}
      closable={false}
      visible={visible}
      handleCancel={handleCancel}
    >
      {src && (
        <Cropper
          src={src}
          ref={cropperRef}
          style={cropperStyle}
          guides={false}
          movable={false}
          zoomable={false}
          autoCropArea={1}
          minContainerWidth={1}
          minContainerHeight={1}
          minCropBoxWidth={minCropSize}
          minCropBoxHeight={minCropSize}
          aspectRatio={aspectRatio || foodBannerAspectRatio}
        />
      )}
      <Button className={styles.button} onClick={handleCancel}>
        {I18n.t('common.cancel')}
      </Button>
      <Button
        className={styles.button}
        onClick={handleConfirm}
        loading={loading}
      >
        {I18n.t('common.confirm')}
      </Button>
    </Modal>
  );
};

ImageEditor.propTypes = {
  src: PropTypes.string,
  onCropImage: PropTypes.func,
  aspectRatio: PropTypes.number,
  creativeType: PropTypes.string,
};

ImageEditor.defaultProps = {
  src: '',
  onCropImage: noop,
  aspectRatio: null,
  creativeType: '',
};

export default ImageEditor;
