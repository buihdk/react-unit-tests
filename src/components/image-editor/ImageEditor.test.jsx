import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ImageEditor from './ImageEditor';

const props = {
  src: 'src',
  onEditedImage: jest.fn(),
};

beforeEach(() => {
  props.src = 'src';
  props.onEditedImage.mockClear();
});

describe('Test Upload Component', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(<ImageEditor {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
