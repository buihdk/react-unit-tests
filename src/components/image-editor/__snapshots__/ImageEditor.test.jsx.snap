// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Upload Component renders without crashing 1`] = `
<Modal
  centered={true}
  closable={false}
  confirmLoading={false}
  footer={null}
  handleCancel={[Function]}
  maskTransitionName="fade"
  okType="primary"
  transitionName="zoom"
  visible={false}
  width="100%"
>
  <ReactCropper
    aspectRatio={2.5}
    autoCropArea={1}
    data={null}
    dragMode="crop"
    enable={true}
    guides={false}
    minContainerHeight={1}
    minContainerWidth={1}
    minCropBoxHeight={40}
    minCropBoxWidth={40}
    movable={false}
    rotateTo={0}
    scaleX={1}
    scaleY={1}
    src="src"
    style={Object {}}
    zoomTo={1}
    zoomable={false}
  />
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
  >
    Cancel
  </Button>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
  >
    Confirm
  </Button>
</Modal>
`;
